docs/backend-requests/making/same-origin.mdx

```diff
@@ -1,5 +1,97 @@
 ---
-sanity_slug: /docs/request-authentication/same-origin
+title: Same-origin requests
+description: Learn about various ways to make same-origin requests to the backend using the Clerk SDK.
 ---
 
-# This is a stub
+# Same-origin requests
+
+If your client and server are on the same origin (e.g. making an API call to `foo.com/api` from JavaScript running on `foo.com`), the [session token](https://clerk.com/docs/backend-requests/resources/session-tokens) is automatically passed to the backend in a cookie. This means that all requests to same-origin endpoints are authenticated by default.
+
+## Using Fetch
+
+You can use the native browser Fetch API as you normally would and the request will be authenticated.
+
+```js
+fetch('/api/foo').then(res => res.json());
+```
+
+## useSWR hook
+
+If you are using React or Next.js and would like to use the [useSWR](https://swr.vercel.app/) hook, it’s as simple as supplying the API route with whichever fetcher function you are using. Because of the [automatic revalidation feature](https://swr.vercel.app/docs/revalidation) of SWR, you need to retrieve and set the session token in the Authorization header. Call the asynchronous `getToken` method from the [`useAuth()`](/docs/references/react/use-auth) hook and add it as a Bearer token.
+
+<CodeBlockTabs options={["index.jsx", "index.tsx"]}>
+```jsx
+import useSWR from 'swr';
+import { useAuth } from '@clerk/nextjs';
+
+export default function useClerkSWR(url) {
+  const { getToken } = useAuth();
+
+  const fetcher = async (...args) => {
+    return fetch(...args, {
+      headers: { Authorization: `Bearer ${await getToken()}` }
+    }).then(res => res.json());
+  };
+
+  return useSWR(url, fetcher);
+}
+```
+
+```tsx
+import useSWR from 'swr';
+import { useAuth } from '@clerk/nextjs';
+
+export default function useClerkSWR(url: string) {
+  const { getToken } = useAuth();
+
+  const fetcher = async (...args: [RequestInfo]) => {
+    return fetch(...args, {
+      headers: { Authorization: `Bearer ${await getToken()}` }
+    }).then(res => res.json());
+  };
+
+  return useSWR(url, fetcher);
+}
+```
+</CodeBlockTabs>
+
+When you re-focus a page or switch between tabs, SWR automatically revalidates data. This can be useful to immediately synchronize to the latest state in scenarios like stale mobile tabs, or laptops that went to sleep.
+
+## Tanstack Query (FKA React Query)
+
+If you are using [Tanstack Query](https://tanstack.com/query/latest), formally known as React Query, in your application, you need to use a query function that throws errors. Since the native Fetch API does not, you can add it in.
```

Comment:
Based on the other similar page:

```suggestion
If you are using [Tanstack Query](https://tanstack.com/query/v4/docs/react/overview), formally known as React Query, in your application, you need to use a query function that throws errors. Since the native Fetch API does not, you can add it in.
```

Reply:
Should we align this with the other page in general? Let's also take Lennart's feedback into consideration here 🙏